test/e2e/ssm/command.go

```diff
@@ -34,12 +34,17 @@ func (s *ssmConfig) runCommandsOnInstance(ctx context.Context, logger logr.Logge
 		logger.Info("ThrottlingException encountered, retrying..")
 		output, err = s.client.SendCommandWithContext(ctx, input)
 	}
+	if err != nil {
+		return nil, fmt.Errorf("got an error calling SendCommand: %w", err)
+	}
 	invocationInput := &ssm.GetCommandInvocationInput{
 		CommandId:  output.Command.CommandId,
 		InstanceId: aws.String(s.instanceID),
 	}
 	// Will wait on Pending, InProgress, or Cancelling until we reach a terminal status of Success, Cancelled, Failed, TimedOut
-	_ = s.client.WaitUntilCommandExecutedWithContext(ctx, invocationInput)
+	_ = s.client.WaitUntilCommandExecutedWithContext(ctx, invocationInput, func(w *request.Waiter) {
+		w.MaxAttempts = 60
```

Comment:
ill add a code comment, this is because upgrade can take longer than the default timeout/retries, esp on rhel.